src/PrestaShopBundle/Resources/config/services/adapter/shop.yml

```diff
@@ -26,6 +26,11 @@ services:
     arguments:
       - "@=service('prestashop.adapter.legacy.context').getContext().link"
 
+  prestashop.adapter.shop.url.product_provider:
+      class: 'PrestaShop\PrestaShop\Adapter\Shop\Url\ProductProvider'
```

Comment:
shouldn't this be in the `data_provider.yml` config file ?

Reply:
and be called `prestashop.adapter.data_provider.product_url` 

(there's already a `prestashop.adapter.data_provider.product` service definition)

Reply:
It's not a data_provider, it's an URLProvider nothing related with our usual DataProvider, this one just generates urls